pkg/conn/mockdb.go

```diff
@@ -0,0 +1,48 @@
+package conn
+
+import (
+	"database/sql"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	check "github.com/pingcap/check"
+
+	"github.com/pingcap/dm/dm/config"
+)
+
+type mockDBProvider struct {
+	verDB *sql.DB // verDB user for show version.
+	db    *sql.DB
+}
+
+// Apply will build BaseDB with DBConfig.
+func (d *mockDBProvider) Apply(config config.DBConfig) (*BaseDB, error) {
+	if err := d.verDB.Ping(); err != nil {
```

Comment:
What if `verDB` is nil? Will we panic here?

Reply:
yes 😂
i copy this func from `testMaster` and in that test, verDB is alway not nil,
but i still need fix  this because this Apply will used in many other tests